plugin/kubernetes/federation.go

```diff
@@ -23,23 +24,30 @@ const (
 
 // Federations is used from the federations plugin to return the service that should be
 // returned as a CNAME for federation(s) to work.
-func (k *Kubernetes) Federations(state request.Request, fname, fzone string) (msg.Service, error) {
-	nodeName := k.localNodeName()
-	node, err := k.APIConn.GetNodeByName(nodeName)
-	if err != nil {
-		return msg.Service{}, err
-	}
-	r, err := parseRequest(state)
-	if err != nil {
-		return msg.Service{}, err
-	}
+func (h *Handler) Federations(state request.Request, fname, fzone string) (msg.Service, error) {
+	for _, k := range h.Kubernetes {
+		zone := plugin.Zones(k.Zones).Matches(state.Name())
+		if zone == "" {
+			continue
+		}
+		nodeName := k.localNodeName()
+		node, err := k.APIConn.GetNodeByName(nodeName)
+		if err != nil {
+			return msg.Service{}, err
+		}
+		r, err := parseRequest(state)
+		if err != nil {
+			return msg.Service{}, err
+		}
 
-	lz := node.Labels[LabelZone]
-	lr := node.Labels[LabelRegion]
+		lz := node.Labels[LabelZone]
+		lr := node.Labels[LabelRegion]
 
-	if r.endpoint == "" {
-		return msg.Service{Host: dnsutil.Join([]string{r.service, r.namespace, fname, r.podOrSvc, lz, lr, fzone})}, nil
-	}
+		if r.endpoint == "" {
+			return msg.Service{Host: dnsutil.Join([]string{r.service, r.namespace, fname, r.podOrSvc, lz, lr, fzone})}, nil
+		}
 
-	return msg.Service{Host: dnsutil.Join([]string{r.endpoint, r.service, r.namespace, fname, r.podOrSvc, lz, lr, fzone})}, nil
+		return msg.Service{Host: dnsutil.Join([]string{r.endpoint, r.service, r.namespace, fname, r.podOrSvc, lz, lr, fzone})}, nil
+	}
+	return msg.Service{}, nil // need to return error here?
```

Comment:
we only returned an empty service with `err != nil`, so I think we should return an error here